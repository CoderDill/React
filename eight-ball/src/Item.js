const Item = ({ id, name, qty }) => {
  return (
    <ul>
      <li>Product Name: {name}</li>
      <li>Product Quantity: {qty}</li>
    </ul>
  );
};

export default Item;
