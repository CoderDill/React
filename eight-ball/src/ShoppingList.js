import { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";
import { v4 as uuid } from "uuid";

const ShoppingList = () => {
  const initial_state = [
    { id: uuid(), name: "Peanut Butter", qty: 2 },
    { id: uuid(), name: "Whole Milk", qty: 1 },
  ];
  const [items, setItems] = useState(initial_state);
  const addItem = (newItem) => {
    setItems((items) => [...items, { ...newItem, id: uuid() }]);
  };
  return (
    <>
      <h3>Shopping List</h3>
      <NewItemForm addItem={addItem} />
      <div>
        {items.map(({ id, name, qty }) => (
          <Item id={id} name={name} qty={qty} key={id} />
        ))}
      </div>
    </>
  );
};

export default ShoppingList;
