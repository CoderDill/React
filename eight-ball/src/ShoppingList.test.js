import { render, fireEvent } from "@testing-library/react";
import Item from "./Item";
import ShoppingList from "./ShoppingList";

it("should render", function () {
  render(<ShoppingList />);
});

it("should match snapshot", function () {
  const { asFragment } = render(<ShoppingList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new item", function () {
  const { queryByText, getByLabelText } = render(<ShoppingList />);
  const input = getByLabelText("Product");
  const btn = queryByText("Add Item");
  expect(queryByText("Product Name: Chocolate Milk")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "Chocolate Milk" } });
  fireEvent.click(btn);
  expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
});
