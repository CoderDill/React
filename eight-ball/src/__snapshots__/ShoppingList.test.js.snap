// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <h3>
    Shopping List
  </h3>
  <form>
    <label
      for="name"
    >
      Product
    </label>
    <input
      id="name"
      name="name"
      qty="qty"
      type="text"
      value=""
    />
    <label
      for="qty"
    >
      Quantity: 
    </label>
    <input
      id="qty"
      max="10"
      min="1"
      name="qty"
      type="range"
      value=""
    />
    <button>
      Add Item
    </button>
  </form>
  <div>
    <ul>
      <li>
        Product Name: Peanut Butter
      </li>
      <li>
        Product Quantity: 2
      </li>
    </ul>
    <ul>
      <li>
        Product Name: Whole Milk
      </li>
      <li>
        Product Quantity: 1
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
